### ISSUE TITLE
Using clsBuffer, an issue when adding a byte after flushing the buffer.

### ISSUE THREAD

**Issue by LostSoulFly** [Aeirex/Eclipse-Worlds#2]
After a Flush of the buffer, if the first thing put into it is a Byte, the buffersize/writehead will get out of sync and cause issues.

I see this problem exists in many other copies of this class, sadly.

I believe I've remedied it here:

In:

```
Public Sub Allocate(ByVal nLength As Long)
    If BufferSize = 0 And nLength > 1 Then nLength = nLength - 1
    BufferSize = BufferSize + nLength
    ReDim Preserve Buffer(0 To BufferSize)
End Sub
```

Should be changed to:

```
Public Sub Allocate(ByVal nLength As Long)
    If BufferSize = 0 And WriteHead > 1 And nLength > 1 Then nLength = nLength - 1
    BufferSize = BufferSize + nLength
    ReDim Preserve Buffer(0 To BufferSize)
End Sub
```

**Comment by abhijeetviswa** [comment]
That's a good catch. I went through the code. Another way to fix that would be to remove: 
```
If BufferSize = 0 And WriteHead > 1 And nLength > 1 Then nLength = nLength - 1
```
I did some testing and there seems to be no different even if the first thing written to the buffer is a byte . The reason why the reason occurs is because the _nLength - 1_ part runs but the WriteHead doesn't keep up. 

Make sure you post a bug fix on http://eclipseorigins.com as well so that the others in the community will know about this issue. This will be patched in Eclipse Worlds.
